### ISSUE TITLE
Cancel previous running compile process in watch mode

### ISSUE THREAD

**Issue by andys8** [rtfeldman/node-test-runner#194]
I've noticed `node-test-runner` started in watch mode is looking for file changes and triggers build and tests. This is not aware of currently running compile or test processes.

If you hit `save` several times, this is what will happen:
![screenshot from 2017-08-02 18-52-24](https://user-images.githubusercontent.com/13085980/28884979-5ba0e7f2-77b4-11e7-8371-9f5daf2ee4ad.png)
![screenshot from 2017-08-02 18-53-18](https://user-images.githubusercontent.com/13085980/28884985-602d297a-77b4-11e7-81c2-e9b7fe55044b.png)

This is for demonstration but it happens for me while developing: Elm-format will run on save and triggers `elm-test` twice. If it's possible, I think it could be a solution to cancel currently executed tasks before starting a new compile process. Or is this a known issue or not possible to recognize?

Issue could be related to #110

**Comment by zwilias** [comment]
I think this should (at least) debounce 🤔

**Comment by andys8** [comment]
@zwilias It would help. If the test execution duration isn't short it will still be painful to watch outdated code being tested ;)

I started digging into the implementation. A node module called "[Chokidar](https://github.com/paulmillr/chokidar)" is used to wrap file system events. I'm curious about the provided options. There is `awaitWriteFinish` which is not activated by default. And there is `awaitWriteFinish.stabilityThreshold` described as follows:
> Amount of time in milliseconds for a file size to remain constant before emitting its event.

Sounds interesting, could be a potential option.

**Comment by rtfeldman** [comment]
@andys8 I would love to accept a PR for this! ❤️

**Comment by andys8** [comment]
Another interesting thing is that `node-test-runner` is using `node-elm-compiler` which is using `node-cross-spawn` to spawn children processes on different platforms, but hasn't the functionality to stop them. 

There is this discussion:
https://github.com/IndigoUnited/node-cross-spawn/issues/40

The result was @kentcdodds creating [`cross-spawn-with-kill`](https://github.com/kentcdodds/cross-spawn-with-kill).

**Comment by andys8** [comment]
@rtfeldman I changed the configuration of `chokidar` and it works for me to solve the problem with `elm-format` triggering `elm-test` in watch mode twice in practice. It will not help if somebody keeps hitting the save button. I think it could also make sense to stop the previous running process with the suggested changes in `node-elm-compiler` and using the `cross-span` fork. It's hard to estimate possible problems due to killing running processes (will they leave files behind?). That's why I added the simple PR which could work. It's your decision :) I'm happy if you try the PR and compare the changes.
